File: 은지/LV2/프린터.py
def solution(priorities, location):
    answer = 0
    idx = [j for j in range(len(priorities))]
    x = priorities.pop(0)
    queue = [x]
    
    while queue:
        v = queue.pop(0)
        i = idx.pop(0)

        if len(priorities):
            max_v = max(priorities)
        else:
            return answer + 1

        if max_v > v:
            priorities.append(v)
            idx.append(i)
            x = priorities.pop(0)
            queue.append(x)
        else:
            if i == location:
                return answer + 1
            else:
                x = priorities.pop(0)
                queue.append(x)
                answer += 1
    return answer

p = [2, 1, 3, 2]
l = 1
print(solution(p, l))

File: 은지/LV2/프린터.js
function solution(priorities, location) {
  var answer = 0;
  var idx = []
  for (let j=0; j<priorities.length; j++) {
    idx.push(j)
  }
  var x = priorities.shift()
  var queue = [x]

  while (queue.length > 0) {
    var v = queue.shift()
    var i = idx.shift()

    if (priorities.length) {
      var max_v = Math.max(...priorities)
    } else {
      return answer + 1
    }
    
    if (max_v > v) {
      priorities.push(v)
      idx.push(i)
      x = priorities.shift()
      queue.push(x)
    } else {
      if (i === location) {
        return answer + 1
      } else {
        x = priorities.shift()
        queue.push(x)
        answer += 1
      }
    }
  }
  return answer;
}